lib/cocoapods-core/source.rb

```diff
@@ -262,7 +262,13 @@ def search(query)
       if query.is_a?(Dependency)
         query = query.root_name
       end
-      found = Pathname.glob(pod_path(query)).map { |path| path.basename.to_s }
+
+      found = []
+      Pathname.glob(pod_path(query)) do |path|
+        next if Dir[path.to_s + '/*'].empty?
```

Comment:
Is `Dir[something].empty?` what @segiddins was trying to avoid? Does this hit the FS? I think in his commit this was not using `Dir`

Reply:
Is there an alternative that works on pre-2.4 ruby's or is there a way to run different code depending on ruby versions? See the stackoverflow question referenced above.

Reply:
ah sorry did not connect the two, yeah I am uncertain we can wait for @segiddins to weigh in.

Reply:
`next if Dir(path).empty?`

Reply:
ugh looks like thats 2.4+ as well.

`next unless Dir.each_child(path).any?` might work? (the docs dont say when each method was introduced)

Reply:
each_child was introduced in 2.5 - https://blog.bigbinary.com/2017/11/21/ruby-2_5-introduces-dir-children-and-dir-each_child.html

Reply:
ugh such terrible docs

maybe `Dir.foreach(path).any? {|child| child != '.' && child != '.. }`

I just really want to avoid globbing whenever possible, as it's expensive